examples/templates/kubernetes/main.tf

```diff
@@ -23,63 +26,81 @@ variable "use_kubeconfig" {
   Set this to true if the Coder host is running outside the Kubernetes cluster
   for workspaces.  A valid "~/.kube/config" must be present on the Coder host.
   EOF
+  default     = false
 }
 
 variable "namespace" {
   type        = string
-  sensitive   = true
   description = "The Kubernetes namespace to create workspaces in (must exist prior to creating workspaces)"
 }
 
-variable "cpu" {
-  description = "CPU (__ cores)"
-  default     = 2
-  validation {
-    condition = contains([
-      "2",
-      "4",
-      "6",
-      "8"
-    ], var.cpu)
-    error_message = "Invalid cpu!"
+data "coder_parameter" "cpu" {
+  name    = "CPU (cores)"
+  default = "2"
+  icon    = "/icon/memory.svg"
+  mutable = true
+  option {
+    name  = "2 Cores"
+    value = "2"
+  }
+  option {
+    name  = "4 Cores"
+    value = "4"
+  }
+  option {
+    name  = "6 Cores"
+    value = "6"
+  }
+  option {
+    name  = "8 Cores"
+    value = "8"
   }
 }
 
-variable "memory" {
-  description = "Memory (__ GB)"
-  default     = 2
-  validation {
-    condition = contains([
-      "2",
-      "4",
-      "6",
-      "8"
-    ], var.memory)
-    error_message = "Invalid memory!"
+data "coder_parameter" "memory" {
+  name    = "Memory (GB)"
+  default = "2"
+  icon    = "/icon/memory.svg"
+  mutable = true
+  option {
+    name  = "2 GB"
+    value = "2"
+  }
+  option {
+    name  = "4 GB"
+    value = "4"
+  }
+  option {
+    name  = "6 GB"
+    value = "6"
+  }
+  option {
+    name  = "8 GB"
+    value = "8"
   }
 }
 
-variable "home_disk_size" {
-  type        = number
-  description = "How large would you like your home volume to be (in GB)?"
-  default     = 10
+data "coder_parameter" "home_disk_size" {
+  name    = "Home Disk Size (GB)"
+  default = "10"
```

Comment:
type: number? Otherwise, validation won't work.

Reply:
Thank you, I may have missed this